### PR TITLE
chore: add attachToContainer methods to connect to the tty of the container

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?

there is one extra commit that is part of https://github.com/containers/podman-desktop/pull/3870

- [x] : depends on https://github.com/containers/podman-desktop/pull/3870

### What issues does this PR fix or reference?

related to https://github.com/containers/podman-desktop/issues/970

### How to test this PR?

Unit tests added

### CODE REVIEW

**Comment by benoitf** [comment]
PR updated as required PR has been merged

File: packages/main/src/plugin/container-registry.ts [hunk 1]
```diff
@@ -1347,6 +1391,135 @@ export class ContainerProviderRegistry {
...
+    openStdin?: boolean,
+  ): Promise<void> {
+    // if option is not specified, try to look if the container is using tty or not
+    if (hasTty === undefined || openStdin === undefined) {
+      const containerInspectInfo = await container.inspect();
+      hasTty = containerInspectInfo.Config.Tty;
+      openStdin = containerInspectInfo.Config.OpenStdin;
+    }
+    // no tty and no stdin, do not need to try to attach a terminal
+    if (!hasTty && !openStdin) {
```
Comment (dgolovin) [inline packages/main/src/plugin/container-registry.ts]: Should the case when hasTty = true and openStdin=false be also banned? It is kind of useless to have terminal enabled without any input attached. Nothing is going to show up in there on itself and no reaction on input ether.

Is there a PR for renderer part to test?

Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts]: I've pushed a fixup commit as yes we should try to keep data only if it is useful.

so, here in this code, we were just trying to attach the output after the start of a container but anyway we were not displaying a way to connect. But better to not keep data if we're not using them.

Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts]: here is the PR to test the renderer part https://github.com/containers/podman-desktop/pull/3900

File: packages/main/src/plugin/container-registry.ts [hunk 2]
```diff
@@ -1099,6 +1136,13 @@ export class ContainerProviderRegistry {
...
     let telemetryOptions = {};
     try {
+      const engine = this.internalProviders.get(engineId);
+      const container = this.getMatchingContainer(engineId, id);
+
+      if (engine) {
+        await this.attachToContainer(engine, container);
+      }
+
       return this.getMatchingContainer(engineId, id).restart();
```
Comment (lstocchi) [inline packages/main/src/plugin/container-registry.ts]: `return container.restart`

**Review by lstocchi — APPROVED** [review]
LGTM

File: packages/main/src/plugin/container-registry.ts (hunk 2 above)
Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts]: fixed 👍 good catch

**Pull request merged**